### PR TITLE
BUG: Fix incorrect _is_scalar_access check in iloc

### PR DESCRIPTION
**Pull request by jbrockmendel**
We only have two test cases that make it to the _take_with_is_copy line in _iget_item_cache

### CODE REVIEW

**Comment by jbrockmendel** [comment]
updated with test, so im now officially calling this a bugfix

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3467,13 +3467,13 @@ def _get_item_cache(self, item):
...
 
-    def _iget_item_cache(self, item):
+    def _iget_item_cache(self, item: int):
         """Return the cached item, item represents a positional indexer."""
         ax = self._info_axis
         if ax.is_unique:
             lower = self._get_item_cache(ax[item])
         else:
-            lower = self._take_with_is_copy(item, axis=self._info_axis_number)
+            return self._ixs(item, axis=1)
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Is this change related to the `_is_scalar_access` fix?

Comment (jbrockmendel) [reply pandas/core/generic.py]: Yes.  With the is_scalar_access fix, this line is now reached, where it fails because `_take_with_is_copy` expects an ndarray, not int.

Comment (jorisvandenbossche) [reply pandas/core/generic.py]: OK!

**Comment by jreback** [comment]
thanks!
